mlir-sycl/lib/Dialect/SYCL/Transforms/ConstantPropagation.cpp

```diff
@@ -1041,26 +1041,37 @@ void ConstantPropagationPass::runOnOperation() {
       return;
     }
 
-    constexpr auto isSingleton = [](auto range) {
-      return std::next(range.begin()) == range.end();
-    };
-
     auto module = op->getParentOfType<ModuleOp>();
-    std::optional<SymbolTable::UseRange> uses =
+    std::optional<SymbolTable::UseRange> optUses =
         SymbolTable::getSymbolUses(op, module);
-    if (!(uses && isSingleton(*uses))) {
+    if (!optUses) {
+      LLVM_DEBUG(llvm::dbgs().indent(2)
+                 << "SYCL constant propagation pass expects a "
+                    "single kernel launch point\n");
```

Comment:
Is this error message up-to-date?

Reply:
Yes, no uses means no `schedule_kernel`, IMO. I think it works

Reply:
```suggestion
                 << "SYCL constant propagation pass expects at "
                    "least one single kernel launch point\n");
```